hazelcast/src/main/java/com/hazelcast/internal/partition/impl/PartitionStateManager.java

```diff
@@ -210,6 +212,36 @@ boolean initializePartitionAssignments(Set<Member> excludedMembers) {
         return true;
     }
 
+    void batchUpdateReplicas(PartitionReplica[][] newState) {
+        PartitionIdSet changedOwnersSet = new PartitionIdSet(partitionCount);
+        for (int partitionId = 0; partitionId < partitionCount; partitionId++) {
+            InternalPartitionImpl partition = partitions[partitionId];
+            PartitionReplica[] replicas = newState[partitionId];
+            if (partition.setReplicas(replicas, false)) {
+                changedOwnersSet.add(partitionId);
+            }
+        }
+        partitionOwnersChanged(changedOwnersSet);
+    }
+
+    /**
+     * Called after a batch of partition replica assignments have been applied. This is an optimization for batch
+     * changes, to avoid repeatedly performing costly computations (like updating partition assignments stamp).
+     * <p><b>
+     * If this logic changes, consider also changing the implementation of
+     * {@link PartitionReplicaInterceptor#replicaChanged(int, int, PartitionReplica, PartitionReplica)}, which should apply
+     * the same logic per partition.
+     * </b></p>
+     *
+     * @param partitionIdSet
+     */
+    void partitionOwnersChanged(PartitionIdSet partitionIdSet) {
+        partitionIdSet.intIterator().forEachRemaining(
+                (IntConsumer) partitionId -> partitionService.getReplicaManager().cancelReplicaSync(partitionId));
+        partitionService.getPartitionStateManager().updateStamp();
```

Comment:
Nit-pick: We can just call `updateStamp();` instead of `partitionService.getPartitionStateManager().updateStamp();` here, no?

Reply:
yup, too many hops for no reason, good spot!
fixed in b222509